includes/action-scheduler/class-init.php

```diff
@@ -0,0 +1,213 @@
+<?php
+namespace um\action_scheduler;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'um\action_scheduler\Init' ) ) {
+
+	/**
+	 * Class Init
+	 *
+	 * @package um\action_scheduler
+	 */
+	class Init {
+
+		protected $enabled = false;
+
+		protected $default_group = 'ultimate-member';
+
+		public function __construct() {
+			$lib_path = UM_PATH . 'includes/lib/action-scheduler/action-scheduler.php';
+			if ( ! file_exists( $lib_path ) ) {
+				wp_die( esc_html__( 'Error: Action Scheduler library is missing.', 'ultimate-member' ) );
```

Comment:
Probably wp_die isn't a great UX, because if the user has some issues with the files he get wp_die for entire website. I think we can make the SiteHealth notification and hide the option to activate Action Scheduler.

Reply:
Check Woocommerce action scheduler also

Reply:
I’ve made a few updates. We now verify if WooCommerce is installed and activated. If it is, we then check for the presence of the Action Scheduler. If neither of these conditions is met, we check whether the Action Scheduler file exists in the Ultimate Member library directory. If everything is in order, the setting is displayed; otherwise, the setting is hidden, and we show a notification.